Userland/Applications/TextEditor/main.cpp

```diff
@@ -72,17 +78,28 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     window->set_icon(app_icon.bitmap_for_size(16));
 
     if (file_to_edit) {
-        FileArgument parsed_argument(file_to_edit);
-        auto response = FileSystemAccessClient::Client::the().request_file_read_only_approved(window, parsed_argument.filename());
+        auto filename = String::from_utf8(StringView(file_to_edit, strlen(file_to_edit)));
+        if (filename.is_error())
+            return 1;
+        FileArgument parsed_argument(filename.value());
+        auto response = FileSystemAccessClient::Client::the().request_file_read_only_approved(
+            window, parsed_argument.filename().to_deprecated_string());
 
         if (response.is_error()) {
             if (response.error().code() == ENOENT)
-                text_widget->open_nonexistent_file(parsed_argument.filename());
+                text_widget->open_nonexistent_file(
+                    parsed_argument.filename().to_deprecated_string());
             else
                 return 1;
         } else {
-            TRY(text_widget->read_file(response.value().filename(), response.value().stream()));
-            text_widget->editor().set_cursor_and_focus_line(parsed_argument.line().value_or(1) - 1, parsed_argument.column().value_or(0));
+            if (text_widget
+                    ->read_file(response.value().filename(),
+                        response.value().stream())
+                    .is_error())
+                return 1;
```

Comment:
Use TRY() instead of manually checking for the error. Also, why is this on so many lines now?